packages/authorization-server/src/services/tokenService.ts

```diff
@@ -109,6 +109,8 @@ export function tokenServiceBuilder({
         verifyClientAssertion(request.client_assertion, request.client_id);
 
       if (clientAssertionErrors) {
+        // TODO double check if errors have to be logged or put inside the error below (check the same for parameters errors)
+        logger.warn(clientAssertionErrors.map((error) => error.detail));
```

Comment:
the error message should already be logged (I don't recall, in `makeApiProblem` maybe), so we can put the errors in the exception and use the standard flow